nats-base-client/headers.ts

```diff
@@ -182,41 +180,74 @@ export class MsgHdrsImpl implements MsgHdrs {
     return k.trim();
   }
 
-  get(k: string): string {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
-    const a = this.headers.get(key);
-    return a ? a[0] : "";
+  keys(): string[] {
+    const keys = [];
+    for (const sk of this.headers.keys()) {
+      keys.push(sk);
+    }
+    return keys;
   }
 
-  has(k: string): boolean {
-    return this.get(k) !== "";
+  findKeys(k: string, exact = false): string[] {
+    if (exact) {
+      return [k];
+    }
+    const lci = k.toLowerCase();
+    const keys = this.keys();
+    return keys.filter((v) => {
+      return lci === v.toLowerCase();
+    });
   }
 
-  set(k: string, v: string): void {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
-    const value = MsgHdrsImpl.validHeaderValue(v);
-    this.headers.set(key, [value]);
+  get(k: string, exact = false): string {
+    const keys = this.findKeys(k, exact);
+    if (keys.length) {
+      const v = this.headers.get(keys[0]);
+      if (v) {
+        return Array.isArray(v) ? v[0] : v;
+      }
+    }
+    return "";
+  }
+
+  has(k: string, exact = false): boolean {
+    return this.findKeys(k, exact).length > 0;
+  }
+
+  set(k: string, v: string, exact = false): void {
+    this.delete(k, exact);
+    this.append(k, v);
   }
 
   append(k: string, v: string): void {
-    const key = MsgHdrsImpl.canonicalMIMEHeaderKey(k);
+    // validate the key
+    MsgHdrsImpl.canonicalMIMEHeaderKey(k);
     const value = MsgHdrsImpl.validHeaderValue(v);
-    let a = this.headers.get(key);
+    let a = this.headers.get(k);
```

Comment:
If you had the `exact` option, you would propagate it to the get() and set() below.

Reply:
changed see new

Reply:
this is old